block/local/adapter.go

```diff
@@ -172,6 +172,24 @@ func (l *Adapter) Get(obj block.ObjectPointer, _ int64) (reader io.ReadCloser, e
 	return f, nil
 }
 
+func (l *Adapter) List(storageNamespace, prefix string) ([]string, error) {
+	p := path.Join(l.path, storageNamespace, prefix)
+	files, err := ioutil.ReadDir(p)
```

Comment:
wrap p with filepath.Clean - or the individuals: storage namespace and prefix

Reply:
Done.